src/main/java/seedu/address/storage/JsonAdaptedClient.java

```diff
@@ -126,17 +129,33 @@ public Client toModelType() throws IllegalValueException {
         }
         final Email modelEmail = new Email(email);
 
+        //check nextMeeting
         if (lastMet == null) {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, LastMet.class.getSimpleName()));
         }
-        final LastMet modelLastMet = ParserUtil.parseLastMet(lastMet);
+        LastMet tempLastMet = ParserUtil.parseLastMet(lastMet);
 
         if (nextMeeting == null) {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
                     NextMeeting.class.getSimpleName()));
         }
-        final NextMeeting modelNextMeeting = ParserUtil.parseNextMeeting(nextMeeting);
-        modelNextMeeting.setWithWho(modelName);
+        Optional<OptionalNonStringBasedField> parsedMeeting = ParserUtil.parseMeeting(nextMeeting);
+        NextMeeting tempNextMeeting = NextMeeting.NULL_MEETING;
+        if (parsedMeeting.get() instanceof LastMet) {
+            LastMet newLastMet = (LastMet) parsedMeeting.get();
+            if (tempLastMet.isEmpty()) {
+                tempLastMet = newLastMet;
+            }
+            if (newLastMet.compareWithDirection(tempLastMet, SortDirection.of("asc")) >= 0) {
+                tempLastMet = newLastMet;
+            }
```

Comment:
Perhaps we can simplify this to
```suggestion
            tempLastMet = tempLastMet.getLaterLastMet(newLastMet);
```